seasketch/geoprocessing
# 7x docs and stability improvements

docs:
- Fix 6.1.0 tutorial, referenced features not available in 6.x.  Also some issues with ordering of steps.
- Publish 7.0 docs, complete with systemsetup, sampleproject and newproject tutorials, as well as a number of guides.
- remove codespaces as documented tutorial option

breaking:
- drop deprecated loadCogWindow and bboxToPixelEdge
- drop overlapRaster
- include dataproviders in top-level export, remove as separate top-level module, migrate base project to use it.
- remove genPreprocessor. Replace with clipToPolygonFeatures, genClipToPolygonFeatures, clipToPolygonDatasources and genClipToPolygonDatasources.
- drop genRandomSketch, merge into genRandomFeature and refactor with more options

deprecations:
- deprecate fgbFetchAll for loadFgb
- deprecate genClipLoader

enhancements:
- add validatePolygon preprocessor, with ensureValidPolygon helper function
- add roundDecimalFormat and numberFormat helper functions
- add toSketchProperties helper function
- add getFeaturesForSketchBboxes function with split feature, add splitBBoxAntimeridian and cleanBBox helpers, 
- make MetricGroup `type` property optional.
- export intersection functions - intersectInChunks, intersectInChunksArea, intersectSum
- add getMetricGroupDatasource and getMetricGroupClassKey methods to projectClientBase
- improve metricGroupSchema in schemas.json

project init:
- drop bbox, planning area, template questions from init
- remove template-ocean-eez from workspace so no longer tested (likely to be removed in future)
- clean up simple and blank reports (function + cards)
- drop ViabilityPage from TabReport and simply inline all report pages
- dataImport: conditional ask additional properties question
- sort init language alphabetically
- add vitest-fetch-mock dep to base project
- migrate to childProperties
- add validatePolygon preprocessor
- add minSize and enforceMinSize options to preprocessors
- drop global datasources from basic-project template except for osm-land and eez-land-union
- remove default objective and metricgroup from project. objectives.json and metrics.json start as empty arrays now.  This is inline with them being opt-in features rather than forced use.

createReport:
- switch vectorFunction to use getFeaturesForSketchBboxes.
- migrate vectorFunction/rasterFunction/OverlapCard from using NullSketch in result to get sketch properties to using useSketchProperties 
- createFunction and createReport now default to async geoprocessing function.  The user often doesn't know which they need right away.

internal:
- rename split.js to antimeridian.js.
- storybook no longer auto-opens web browser, caused error in geoprocessing WSL.
- migrate to loadFgb internally from deprecated fgbFetchAll
- upgrade to flatgeobuf 3.36.0 and drop use of web streams polyfill and node-fetch now that bug fixed.
- add Vite babel plugins for storybook
- Remove antimeridian tests from getFeatures
- add getFgbPath method to projectClientBase for loading of flatgeobuf from local filesystem
- move all createReport assets over to now unused template-ocean-eez directory. eventually need to simplify this.

bugs:
- fix genRandomPolygon name bug
- fix i18nAsync bug using base english strings instead of letting component render its own
- fix storybook error when sketch name has spaces or any characters not valid in a Javascript variable name
- remove .story-cache folders from dist build
- genRandomPolygon - start id's at 1, was causing a bug downstream with 0 id value

upgrade:
- remove old genRandom scripts on upgrade
